### ISSUE TITLE
Conv1DTranspose: Invalid Symbolic Shape + Runtime Crash When output_padding ≥ strides

### ISSUE THREAD

**Issue by MilkFiish** [keras-team/keras#22428]
Conv1DTranspose returns a non-empty symbolic shape (no warnings) when output_padding ≥ strides, but crashes with a cryptic error during eager execution—symbolic/eager shape inference results are inconsistent.
#### Reproducible Code & Results
keras: 3.13.2
```
import keras
import numpy as np

eager_input = np.random.randn(4, 10, 128)
symbolic_input = keras.Input(shape=(10,128))

layer = keras.layers.Conv1DTranspose(
    filters=16,
    kernel_size=3,
    strides=1,
    output_padding=1
)

try:
  eager_shape = layer(eager_input).shape
  print(eager_shape)
except Exception as e:
  print(e)

try:
  symbolic_shape = layer(symbolic_input).shape
  print(symbolic_shape)
except Exception as e:
  print(e)
```
eager_mode results:
```
Exception encountered when calling Conv1DTranspose.call().

{{function_node __wrapped__Conv2DBackpropInput_device_/job:localhost/replica:0/task:0/device:CPU:0}} Conv2DCustomBackpropInput: Size of out_backprop doesn't match computed: actual = 10, computed = 11 spatial_dim: 2 input: 13 filter: 3 output: 10 stride: 1 dilation: 1 [Op:Conv2DBackpropInput] name: 

Arguments received by Conv1DTranspose.call():
  • inputs=tf.Tensor(shape=(4, 10, 128), dtype=float32)
```
symbolic_mode results:
```
(None, 13, 16)
```
#### Colab Link:
https://colab.research.google.com/drive/1SigkueLigpHDiuJUhQ-Q5H2_KUGqG6Jx?usp=sharing

**Comment by dhantule** [comment]
Hi @MilkFiish, thanks for reporting this. 

I was able to reproduce this issue with `Keras 3.13.2` in this [gist](https://colab.sandbox.google.com/gist/dhantule/a77515bf303551d05423542e60b3c444/conv1dtranspose-invalid-symbolic-shape-runtime-crash-when-output_padding-strides-22428.ipynb). We'll look into this and update you.

**Comment by ChiragSW** [comment]
Hi @dhantule, the issue can be solved with a simple fix: 
adding validation in BaseConvTranspose.__init__ that would raise ValueError when any output_padding value is >= stride in the corresponding dimension.
Can I put up a PR for this?

**Comment by ChiragSW** [comment]
The issues: #22429 and #22430 can also be covered in this PR, as I would add cases for all three. The real problem was ```BaseConvTranspose.__init__``` accepted any output_padding value without checking the constraint that output_padding must be less than strides per dim

**Comment by dhantule** [comment]
> Hi [@dhantule](https://github.com/dhantule), the issue can be solved with a simple fix: adding validation in BaseConvTranspose.**init** that would raise ValueError when any output_padding value is >= stride in the corresponding dimension. Can I put up a PR for this?

@ChiragSW, feel free to raise a PR with the fix. thank you.

**Comment by ChiragSW** [comment]
@dhantule the issue has been fixed in my PR, please review

**Comment by hertschuh** [comment]
Fixed by https://github.com/keras-team/keras/pull/22439
